hi-ml-histopathology/src/histopathology/utils/plots_utils.py

```diff
@@ -0,0 +1,253 @@
+        tile_size: int = 224,
+        num_columns: int = 4,
+        figsize: Tuple[int, int] = (10, 10),
+        conf_matrix: Optional[ConfusionMatrix] = None,
+        class_names: Optional[Sequence[str]] = None,
+    ) -> None:
+        """_summary_
+
+        :param plot_options: A set of plot options to produce the desired plot outputs.
+        :param level: Magnification at which tiles are available (e.g. PANDA levels are 0 for original,
+            1 for 4x downsampled, 2 for 16x downsampled). Default 1.
+        :param tile_size: _description_, defaults to 224
+        :param num_columns: Number of columns to create the subfigures grid, defaults to 4
+        :param figsize: The figure size of tiles attention plots, defaults to (10, 10)
+        :param conf_matrix: The confustion metric to be used that depends on the number of classes, defaults to None
+        :param class_names: List of class names, defaults to None
+        :param slides_dataset: The slides dataset from where to load the whole slide images, defaults to None
+        """
+        self.plot_options = validate_plot_options(plot_options)
+        self.class_names = validate_class_names_for_plot_options(class_names, plot_options)
+        self.level = level
+        self.tile_size = tile_size
+        self.num_columns = num_columns
+        self.figsize = figsize
+        self.conf_matrix: ConfusionMatrix = conf_matrix
+        self.slides_dataset: Optional[SlidesDataset]
+
+    def save_slide_node_figures(
+        self, case: str, slide_node: SlideNode, outputs_dir: Path, results: ResultsType
+    ) -> None:
+        """Plots and saves all slide related figures, e.g., `TOP_BOTTOM_TILES` and `SLIDE_THUMBNAIL_HEATMAP`"""
+
+        case_dir = make_figure_dirs(subfolder=case, parent_dir=outputs_dir)
+
+        if PlotOptionsKey.TOP_BOTTOM_TILES in self.plot_options:
+            save_top_and_bottom_tiles(
+                case=case,
+                slide_node=slide_node,
+                figures_dir=case_dir,
+                num_columns=self.num_columns,
+                figsize=self.figsize,
+            )
+        if PlotOptionsKey.SLIDE_THUMBNAIL_HEATMAP in self.plot_options:
+            assert self.slides_dataset
+            save_slide_thumbnail_and_heatmap(
+                case=case,
+                slide_node=slide_node,
+                figures_dir=case_dir,
+                results=results,
+                slides_dataset=self.slides_dataset,
+                tile_size=self.tile_size,
+                level=self.level,
+            )
+
+    def log_slide_plot_options(self) -> None:
+        """Log info for slide plot options."""
+        if PlotOptionsKey.TOP_BOTTOM_TILES in self.plot_options:
+            logging.info("Plotting top and bottom tiles ...")
+
+        if PlotOptionsKey.SLIDE_THUMBNAIL_HEATMAP in self.plot_options:
+            logging.info("Plotting slide thumbnails and heatmaps ...")
+
+    def save_all_plot_options(
```

Comment:
[Minor] This method name is a bit confusing to me (i.e. saving the plots or the options?). How about `save_plots()`?

Reply:
done!